stonesoup/metricgenerator/tests/test_uncertaintymetric.py

```diff
@@ -0,0 +1,92 @@
+"""Uncertainty metric tests."""
+import datetime
+
+import numpy as np
+
+from ..manager import SimpleManager
+from ..uncertaintymetric import UncertaintyMetric
+from ...types.detection import Detection
+from ...types.groundtruth import GroundTruthPath, GroundTruthState
+from ...types.state import State, GaussianState
+from ...types.track import Track
+
+
+def test_uncertaintymetric_extractstates():
+    """Test uncertainty metric extract states."""
+    generator = UncertaintyMetric()
+
+    # Test state extraction
+    time_start = datetime.datetime.now()
+    detections = [Detection(state_vector=np.array([[i]]), timestamp=time_start)
+                  for i in range(5)]
+    tracks = {Track(states=[State(state_vector=[[i]],
+                                  timestamp=time_start)]) for i in range(5)}
+    truths = {GroundTruthPath(states=[GroundTruthState(state_vector=[[i]],
+                                                       timestamp=time_start)])
+              for i in range(5)}
+
+    det_states = generator.extract_states(detections)
+    assert det_states.states == detections
+    track_states = generator.extract_states(tracks)
+    assert set(track_states) == set(t.states[0] for t in tracks)
+    truth_states = generator.extract_states(truths)
+    assert set(truth_states) == set(t.states[0] for t in truths)
+
+
+def test_uncertaintymetric_compute_uncertainty():
+    """Test uncertainty metric compute uncertainty."""
+    generator = UncertaintyMetric()
+
+    time = datetime.datetime.now()
+    track = Track(states=[GaussianState(state_vector=[[1], [2], [1], [2]],
+                                        timestamp=time,
+                                        covar=np.diag([i, i, i, i]))
+                          for i in range(5)])
+
+    metric = generator.compute_uncertainty(track.states)
+
+    assert metric.title == "Uncertainty Sum"
+    assert metric.value == 20
+    assert metric.timestamp == time
+    assert metric.generator == generator
+
```

Comment:
Adding a test to check whether an error is thrown when the states list passed in to the `compute_uncertainty` method contains more than one timestamp will increase code coverage.